programming-extensions/programming-extension-pnp/src/main/java/org/objectweb/proactive/extensions/pnp/PNPRemoteObjectFactoryBackend.java

```diff
@@ -230,7 +231,7 @@ public InternalRemoteRemoteObject createRemoteObject(RemoteObject<?> remoteObjec
             }
 
             URI uri = new URI(this.getProtocolId(),
-                              null,
+                              ProActiveInet.getPublicAddress(),
```

Comment:
I think we should explicitly indicate here that the public IP address is intentionally given through the "UserInfo" field and explain why (namespace issue using the 'query' field ?).